pyro/infer/trace_crps.py

```diff
@@ -0,0 +1,94 @@
+from collections import OrderedDict
+
+import torch
+
+import pyro
+import pyrp.poutine as poutine
+from pyro.infer.elbo import ELBO
+from pyro.infer.util import torch_item
+from pyro.util import warn_if_nan
+
+
+class Trace_CRPS:
+    """
+    Posterior predictive CRPS loss.
+
+    This is a non-Bayesian, likelihood-free method; no densities are evaluated.
+    """
+    def __init__(self,
+                 num_particles=2,
+                 max_plate_nesting=float('inf')):
+        assert num_particles >= 2
+        self.max_plate_nesting = max_plate_nesting
+        self.vectorize_particles = True
+
+    def _get_samples(self, model, guide, *args, **kwargs):
```

Comment:
@eb8680 thanks for explaining how to draw posterior predictive samples on Friday! Can you review my use of `uncondition` here? I'm especially unsure whether `tr2` will record the observations.

Reply:
Hmm, I don't think so, `tr2` will end up with samples rather than the observation here.